CDP4Composition/Mvvm/Types/DisposableReactiveList.cs

```diff
@@ -0,0 +1,261 @@
+// <copyright file="DisposableReactiveList.cs" company="RHEA System S.A.">
+//   Copyright (c) 2015-2020 RHEA System S.A.
+// </copyright>
+// -----------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Mvvm.Types
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// Special type of <see cref="ReactiveList{T}"/> that is capable of disposing of child items when items are removed from the list
+    /// Normal "remove" methods are overwritten and made Obsolete, so a developer is forced to make a choice between "remove and dispose", or "remove without dispose".
+    /// </summary>
+    public class DisposableReactiveList<T> : ReactiveList<T> where T : IDisposable
+    {
+        /// <summary>
+        /// Overrides the method implementation in <see cref="ReactiveList{T}"/> and adds an <see cref="ObsoleteAttribute"/>
+        /// so the code doesn't compile anymore when the method is unexpectedly being used.
+        /// </summary>
+        [Obsolete("Clear is deprecated, please use Clear(bool dispose) instead.", true)]
+        public new void Clear()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Wrapper for executing <see cref="Clear(bool)"/> method WITH automatic disposal of concerning objects of type <see cref="T"/>
+        /// </summary>
+        public void ClearAndDispose()
+        {
+            this.Clear(true);
+        }
+
+        /// <summary>
+        /// Wrapper for executing <see cref="Clear(bool)"/> method WITHOUT automatic disposal of concerning objects of type <see cref="T"/>
+        /// </summary>
+        public void ClearWithoutDispose()
+        {
+            this.Clear(false);
+        }
+
+        /// <summary>
+        /// Clear all items of type <see cref="T"/> from this instance
+        /// </summary>
+        /// <param name="dispose">States if the removed objects of type <see cref="T"/> should also be disposed on removal</param>
+        private void Clear(bool dispose)
+        {
+            if (dispose)
+            {
+                foreach (var item in this)
+                {
+                    this.TryDispose(item);
+                }
+            }
+
+            base.Clear();
+        }
+
+        /// <summary>
+        /// Overrides the method implementation in <see cref="ReactiveList{T}"/> and adds an <see cref="ObsoleteAttribute"/>
+        /// so the code doesn't compile anymore when the method is unexpectedly being used.
+        /// </summary>
+        [Obsolete("RemoveAll(IEnumerable<T> items) is deprecated, please use RemoveAll(IEnumerable<T> items, bool dispose) instead.", true)]
+        public new void RemoveAll(IEnumerable<T> items)
+        {
+            throw new NotImplementedException();
```

Comment:
i think this should be a `NotSupportedException`

Reply:
I concur.